### ISSUE TITLE
how to connect.

### ISSUE THREAD

**Issue by rampa069** [michalhol/ps2kbdlib#2]
Hi, How must i connect the keyboard to ESP32? has to be a PS2 keyboard or can be an USB one?

**Comment by cubicap** [comment]
It has to be PS2. Some USB may communicate using the same protocol but I think it is not very probable.

**Comment by rampa069** [comment]
ok. and the pins have some pullup or something or it is straight pin to keyboard?

**Comment by cubicap** [comment]
You have to power the keyboard with 5V source and the keyboard needs 5V communication. So you have to connect level shifter in between the keyboard (5 V) and the ESP (3.3 V). ESP32 has internal pullup resistors so you don't need to worry about them.

**Comment by Isopodus** [comment]
Hi,
i have tried to use your library with your example and also with a level converter such as [this](https://ru.aliexpress.com/item/IIC-I2C-Logic-Level-Converter-Bi-Directional-Module-5V-to-3-3V-For-Arduino/32589088559.html) one, but library didn't work, nothing appears in serial monitor. Can you please make a better explanation of wiring? I think you should also add it to README, btw.
Thanks!

**Comment by cubicap** [comment]
I tried this and it worked well. Do you have everything connected properly? I will update the README to specify available pins better. What board do you use? Also, baud rate of the serial in the example is 115200 opposed to often used 9600, make sure you have set serial terminal/monitor to that.

**Comment by Isopodus** [comment]
Yeap, baud is 115200, maybe i made some mistakes in level converter wiring, i'll try again now. My board is DOIT ESP32 Dev kit (ESP-WROOM-32).

**Comment by Isopodus** [comment]
This is how does my wiring looks like now. Connector in the top goes to ps/2 female connector, than to the keyboard. Yellow wire is CLK, white is DATA. Keyboard blinks when power comes, so ps/2 wiring is probably okay, i also tried to swap yellow and white, but this did not help, it doesn't work.
![IMG_20190507_191412](https://user-images.githubusercontent.com/29857052/57315523-6e883300-70fc-11e9-942c-8d8ea5bd52d9.jpg)
Keyboard is okay too, i tried it with common PS2Keyboard library on Arduino Nano, it worked.

**Comment by cubicap** [comment]
The wiring seems good to me. I am a bit busy right now. I will try to run that at the weekend and let you know.

**Comment by Isopodus** [comment]
Okay, thanks

**Comment by Isopodus** [comment]
Hi again. Did you tried or not yet?

**Comment by cubicap** [comment]
Hi, I just wired it up and everything ran as normal. Only thing I did was changing the pins to 32 data and 33 clock. 
![IMG_20190512_110824](https://user-images.githubusercontent.com/29575279/57580215-3ccdee00-74a7-11e9-89d8-d356932c9ede.jpg)
Only differences I see is that I use USB 5V to power the keyboard and so it has common ground with the ESP and that I have different ESP32 dev kit (something like Espressif ESP32 dev module v2).
